src/main/java/org/opensearch/ad/rest/RestValidateAnomalyDetectorAction.java

```diff
@@ -0,0 +1,144 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ *
+ * Modifications Copyright OpenSearch Contributors. See
+ * GitHub history for details.
+ */
+
+package org.opensearch.ad.rest;
+
+import static org.opensearch.ad.util.RestHandlerUtils.TYPE;
+import static org.opensearch.ad.util.RestHandlerUtils.VALIDATE;
+import static org.opensearch.common.xcontent.XContentParserUtils.ensureExpectedToken;
+
+import java.io.IOException;
+import java.util.List;
+import java.util.Locale;
+
+import org.opensearch.ad.AnomalyDetectorPlugin;
+import org.opensearch.ad.common.exception.ADValidationException;
+import org.opensearch.ad.constant.CommonErrorMessages;
+import org.opensearch.ad.model.AnomalyDetector;
+import org.opensearch.ad.model.DetectorValidationIssue;
+import org.opensearch.ad.model.DetectorValidationIssueType;
+import org.opensearch.ad.model.ValidationAspect;
+import org.opensearch.ad.settings.EnabledSetting;
+import org.opensearch.ad.transport.ValidateAnomalyDetectorAction;
+import org.opensearch.ad.transport.ValidateAnomalyDetectorRequest;
+import org.opensearch.ad.transport.ValidateAnomalyDetectorResponse;
+import org.opensearch.client.node.NodeClient;
+import org.opensearch.cluster.service.ClusterService;
+import org.opensearch.common.ParsingException;
+import org.opensearch.common.settings.Settings;
+import org.opensearch.common.xcontent.XContentParseException;
+import org.opensearch.common.xcontent.XContentParser;
+import org.opensearch.rest.BaseRestHandler;
+import org.opensearch.rest.BytesRestResponse;
+import org.opensearch.rest.RestChannel;
+import org.opensearch.rest.RestRequest;
+import org.opensearch.rest.RestStatus;
+import org.opensearch.rest.action.RestToXContentListener;
+
+import com.google.common.collect.ImmutableList;
+
+/**
+ * This class consists of the REST handler to validate anomaly detector configurations.
+ */
+public class RestValidateAnomalyDetectorAction extends AbstractAnomalyDetectorAction {
+    private static final String VALIDATE_ANOMALY_DETECTOR_ACTION = "validate_anomaly_detector_action";
+
+    public RestValidateAnomalyDetectorAction(Settings settings, ClusterService clusterService) {
+        super(settings, clusterService);
+    }
+
+    @Override
+    public String getName() {
+        return VALIDATE_ANOMALY_DETECTOR_ACTION;
+    }
+
+    @Override
+    public List<Route> routes() {
+        return ImmutableList
+            .of(
+                new Route(
+                    RestRequest.Method.POST,
+                    String.format(Locale.ROOT, "%s/%s", AnomalyDetectorPlugin.AD_BASE_DETECTORS_URI, VALIDATE)
+                ),
+                new Route(
+                    RestRequest.Method.POST,
+                    String.format(Locale.ROOT, "%s/%s/{%s}", AnomalyDetectorPlugin.AD_BASE_DETECTORS_URI, VALIDATE, TYPE)
```

Comment:
Cannot comment on https://github.com/opensearch-project/anomaly-detection/pull/231#discussion_r721536978

Will reply here.

> Do you think it's fair to completely remove the params for just the blocker release due to the fact that /_validate/detector will have the same output as /_validate, or to keep it to give user some information about the type of validation they are doing and make the addition of model validation a smoother experience later on?

It is fine to keep this end point.  I guess Tyler is suggesting not to expose the model type and only deal with detector type before you release non-blocker validation.

> Why does model run both detector/blocker validation and model/non-blocker validation? Shouldn't it just run the latter? If users wanted to run both aspects, they could just pass both types (e.g., <path>/_validate/detector,model)

I like Tyler's suggestion.  detector is for blocker part and model is for non-blocker part.  Profile/stats API use similar format.

Reply:
This makes sense to me, changed the implementation slightly so its similar to Profile API. Detector param for blocker part and Model Param will be for non-blocker part. However, to hide the model type at the moment for 1.2 release I removed it from the ValidationAspect class. So currently validation type can be left empty and it defaults to detector or if its anything else except configs I return an exception stating that the given validation type doesn't exist.